src/main/java/randoop/sequence/SequenceCollection.java

```diff
@@ -227,6 +236,34 @@ public SimpleList<Sequence> getSequencesForType(
       }
     }
 
+    // If we found no sequences of the needed type, use demand-driven input creation to find one
+    // if enabled.
+    if (resultList.isEmpty() && GenInputsAbstract.demand_driven && useDemandDriven) {
+      Log.logPrintf("DemandDrivenInputCreation will try to find a sequence for type %s%n", type);
+      SimpleList<Sequence> sequencesForType;
+      try {
+        // This isn't thread-safe.
+        useDemandDriven = false;
+        sequencesForType =
+            DemandDrivenInputCreation.createInputForType(this, type, exactMatch, onlyReceivers);
```

Comment:
You could also set useDemandDriven to false in the catch statement

Reply:
Thanks for the suggestion! I just want to confirm if I understand correctly: Is the intention behind your suggestion to ensure that, if an exception occurs during demand-driven input creation, it should prevent any further demand-driven calls for the remainder of the run?

Reply:
I misunderstood the purpose of the useDemandDriven variable. Thanks for the clarification. It is fine how it is now.